File: ViewFinderAnywhere/src/main/java/com/fezrestia/android/lib/firebase/FirebaseMessagingServiceImpl.java
package com.fezrestia.android.lib.firebase;

import android.widget.Toast;

import com.fezrestia.android.lib.util.log.Log;
import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;

import java.util.Objects;

import androidx.annotation.NonNull;

public class FirebaseMessagingServiceImpl extends FirebaseMessagingService {
    public static final String TAG = "FirebaseMessagingServiceImpl";

    @SuppressWarnings("PointlessBooleanExpression")
    private static final boolean IS_DEBUG = false | Log.IS_DEBUG;

    @Override
    public void onMessageReceived(@NonNull RemoteMessage remoteMessage) {
        if (IS_DEBUG) Log.logDebug(TAG, "onMessageReceived : E");

        //TODO: Handle message.
        Toast.makeText(
                this,
                "Msg = " + Objects.requireNonNull(remoteMessage.getNotification()).getBody(),
                Toast.LENGTH_SHORT)
                .show();
        if (IS_DEBUG) {
            Log.logDebug(TAG, "### From = " + remoteMessage.getFrom());
            Log.logDebug(TAG, "### Msg  = " + Objects.requireNonNull(remoteMessage.getNotification()).getBody());
        }

        if (IS_DEBUG) Log.logDebug(TAG, "onMessageReceived : X");
    }
}

File: ViewFinderAnywhere/src/main/java/com/fezrestia/android/lib/firebase/FirebaseMessagingServiceImpl.kt
package com.fezrestia.android.lib.firebase

import android.widget.Toast

import com.fezrestia.android.lib.util.log.Log
import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage

class FirebaseMessagingServiceImpl : FirebaseMessagingService() {

    override fun onMessageReceived(remoteMessage: RemoteMessage) {
        if (IS_DEBUG) Log.logDebug(TAG, "onMessageReceived : E")

        val notification = remoteMessage.notification
        val msg = if (notification != null) {
            notification.body ?: "RemoteMessage.Notification.Body is null"
        } else {
            "RemoteMessage.Notification is null"
        }

        //TODO: Handle message.
        Toast.makeText(
                this,
                "Msg = $msg",
                Toast.LENGTH_SHORT)
                .show()
        if (IS_DEBUG) {
            Log.logDebug(TAG, "### From = ${remoteMessage.from}")
            Log.logDebug(TAG, "### Msg  = $msg")
        }

        if (IS_DEBUG) Log.logDebug(TAG, "onMessageReceived : X")
    }

    companion object {
        const val TAG = "FirebaseMessagingServiceImpl"

        private const val IS_DEBUG = false or Log.IS_DEBUG
    }
}
